### ISSUE TITLE
Controls: Inconsistent focus ring rendering

### ISSUE THREAD

**Issue by ItsJonQ** [ItsJonQ/g2#110]
<img width="270" alt="Screen Shot 2020-11-14 at 11 46 10 AM" src="https://user-images.githubusercontent.com/2322354/99152345-322d7f80-266f-11eb-8b12-9874a473516a.png">

Recently, we've made the focus "ring" a `0.5px` box-shadow for the controls. Today, I noticed some strange/inconsistent rendering with these new styles (see screenshot above). Some of the sides appear to be "cut off".

Demo link:
https://g2-components.xyz/?path=/story/designtools-typographytools--bordered-color-panel-expand

It doesn't seem to be affecting the components here:
https://g2-components.xyz/?path=/story/designtools-presentation-components--input-text-field

I'm not entirely sure what may be causing it. There doesn't seem to be any `overflow` based cropping in the UI that's being affected - at least, not from what I can tell.

---

This may be something to keep an eye on.

If this continues to be a problem, and we're adamant on having `1.5px` sub-pixel value borders, an alternative solution may be to render the `box-shadow` as `inset`. From my tests, this seems to resolve the issue. However, this does affect the "roundness" of the corners **ever so slightly**.

---

cc'ing @jasmussen

**Comment by ItsJonQ** [comment]
Poked the UI a bit. Changing the focus shadow value to `1px` resolves the rendering problem.
There's something off when these controls rendering in a specific configuration of `grid` and `flex` 🤔 

(Note: I've experienced random odd CSS behaviour for `grid` and `flex` as well, unrelated to this.)

**Comment by jasmussen** [comment]
It's the `overflow: hidden auto;` applied to the scroll container. Do we need that?

![cropped](https://user-images.githubusercontent.com/1204802/99367559-65178380-28ba-11eb-8c01-0d7fe6580019.gif)

**Comment by ItsJonQ** [comment]
> applied to the scroll container. Do we need that?

Short answer, **yes**.

The overflow scroll handling is part of `Card` by default. It's something that can disabled.
That aside, the Gutenberg sidebar will have overflow handling (which is how it gets it's scrollbar), so this may happen there as well.

With that being said...

If this is problematic within Gutenberg, would you be okay with a "plan B" of either:

1. Using `0.5px inset`
2. Using `1px` instead of sub-pixel rendering

**Comment by jasmussen** [comment]
I have a bit on my plate so I can't dig in too deep, but it seems to me that some overflow handling of the problematic components can address this. It seems like the card should be allowed to handle overflow, the weird thing is that it affects the components themselves, so it seems like there's some inheritance going on that can be unset.

**Comment by ItsJonQ** [comment]
@jasmussen I'll dig into this. I'm really not sure what may be happening. I don't think it's an inheritance thing. (I guess we'll see!)

**Comment by ItsJonQ** [comment]
Definitely a strange (buggy) rendering issue. Check it out!

In the screenshot below, I've forced the focus styles on a bunch of inputs.
In this setup, only the **first** input has the clipping issue. The rest of them render fine.

<img width="330" alt="Screen Shot 2020-11-18 at 9 06 42 AM" src="https://user-images.githubusercontent.com/2322354/99540351-63fb5a80-297d-11eb-9654-e38780bb6feb.png">

Once I start deactivating the focus, the buggy "clipped" ring still only affects the first rendered input:
<img width="315" alt="Screen Shot 2020-11-18 at 9 07 49 AM" src="https://user-images.githubusercontent.com/2322354/99540539-9efd8e00-297d-11eb-8bbf-1408ae249ec5.png">

**Comment by jasmussen** [comment]
In some of my debugging, it was hard for me in the inspector to debug what actually had focus. Usually it's the input directly, but here it seemed like focus was handled differently. I wonder if that plays a role?

**Comment by ItsJonQ** [comment]
@jasmussen ... you won't believe it.

`z-index: 1`

That's giving Chrome a hard time 🙈

**Comment by ItsJonQ** [comment]
Update: Oh my goodness.

I think I found the issue.

It's all the things above... but ONLY... when there's a custom styled `input[type="range"]` being rendered on screen.

Check this out:
https://d.pr/v/bNgUTX

In this video, I forced the focus rings to show. The moment I remove the `input[type="range"]` from the screen, they render fine.

Next steps: Figure out what it is about the `<Slider />` component styles that's making Chrome unhappy.
